config/routes.rb

```diff
@@ -2,7 +2,22 @@
   get 'users', to: 'users#index'
   get 'users/:id', to: 'users#show', as: 'user'
   get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get 'users/:user_id/posts/new', to: 'posts#new', as: 'new_user_post'
   get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  post 'users/:user_id/posts', to: 'posts#create', as: 'create_user_post'
+
+  resources :posts, only: %i[new create] do
+    post 'create_comment', on: :member
+    member do
+      post 'like'
+    end
+  end
+
+  resources :comments, only: %i[new create]
```

Comment:
- [Optional] I would suggest having your `like` and `comment` routes inside the User->Post controllers. Since both belong to a post and the post belongs to a user it would make science to nest your routes inside user and post resources.